src/tests/ftest/util/server_utils.py

```diff
@@ -1052,3 +655,436 @@
+        """Get a test pool object configured to fit the current server config.
+
+        The TestPool parameters returned by this method will be configure a
+        TestPool object with a size (SCM or SCM + NVMe) that does not exceed the
+        specified requirements (scm_ratio, nvme_ratio, and the pool quantity):
+            nvme_ratio > 0:
+                - the pool is configured with both SCM and NVMe
+                - the pool NVMe size equal to the largest 1GiB increment * the
+                  number of targets per engine that does not exceed the
+                  nvme_ratio of the smallest total NVMe capacity of each engine.
+                - if needed the number of targets per engine will be reduced to
+                  meet this size requirement, down to min_targets.
+                - the pool SCM size will be defined by the scm_ratio - as a
+                  precentage of the NVMe size
+            nvme_ratio == 0:
+                - the pool is configured with SCM only
+                - the pool SCM size is set to the scm_ratio of the smallest
+                  total SCM capacity of each engine
+
+        Args:
+            scm_ratio (int, optional): when creating a pool with NVMe
+                (nvme_ratio > 0) this defines the pool SCM size as a percentage
+                of the pool NVMe size. When creating a pool without NVMe
+                (nvme_ratio == 0) this defines the pool SCM size as a percentage
+                of the available SCM storage. Defaults to 6.
+            nvme_ratio (int, optional): this defines the pool NVMe size as a
+                percentage of the available NVMe storage, e.g. 80. If set to 0
+                the pool is created with SCM only. Defaults to 90.
+            min_targets (int, optional): the minimum number of targets per
+                engine that can be configured. Defaults to 1.
+            quantity (int, optional): Number of pools to account for in the size
+                calculations. The pool size returned is only for a single pool.
+                Defaults to 1.
+
+        Raises:
+            ServerFailed: if there was a problem obtaining auto-sized TestPool
+                parameters.
+
+        Returns:
+            dict: the parameters for a TestPool object.
+
+        """
+        # Use the default scm_ratio if one is not provided
+        if scm_ratio is None:
+            scm_ratio = 6
+
+        pool_msg = "{} pool{}".format(quantity, "s" if quantity > 1 else "")
+        self.log.info(
+            "Autosizing TestPool parameters for %s with a %s%% "
+            "scm_ratio and a %s%% nvme_ratio:", pool_msg, scm_ratio, nvme_ratio)
+
+        # Determine the largest SCM and NVMe pool sizes can be used with this
+        # server configuration with an optionally applied ratio.
+        try:
+            available_storage = self.get_available_storage()
+        except ServerFailed as error:
+            raise ServerFailed(
```

Comment:
(pylint-raise-missing-from) Consider explicitly re-raising using the 'from' keyword